pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java

```diff
@@ -636,6 +640,15 @@ public CompletableFuture<Optional<String>> selectAsync(ServiceUnitId bundle,
                             return Optional.empty();
                         }
                         Set<String> candidateBrokers = availableBrokerCandidates.keySet();
+                        Set<String> sheddingExcludedNamespaces = conf.getLoadBalancerSheddingExcludedNamespaces();
+
+                        var namespace = NamespaceBundle.getBundleNamespace(bundle.toString());
+                        if (sheddingExcludedNamespaces.contains(namespace)) {
+                            if (debug(conf, log)) {
+                                log.info("Use round robin broker selector for {}", bundle);
+                            }
+                            return roundRobinBrokerSelectionStrategy.select(candidateBrokers, bundle, context);
+                        }
                         return getBrokerSelectionStrategy().select(candidateBrokers, bundle, context);
```

Comment:
can we pass `namespace` in `getBrokerSelectionStrategy` and return the correct strategy based on the namespace?